Userland/truncate.cpp

```diff
@@ -0,0 +1,106 @@
+#include <LibCore/CArgsParser.h>
+
+#include <stdio.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+enum TruncateOperation {
+    OP_Set,
+    OP_Grow,
+    OP_Shrink,
+};
+
+int main(int argc, char** argv)
+{
+    CArgsParser args_parser("truncate");
+
+    args_parser.add_arg("s", "size", "Resize the target file to (or by) this size. Prefix with + or - to expand or shrink the file, or a bare number to set the size exactly.");
+    args_parser.add_arg("r", "reference", "Resize the target file to match the size of this one.");
+    args_parser.add_required_single_value("file");
+
+    CArgsParserResult args = args_parser.parse(argc, argv);
+
+    if (!args.is_present("s") && !args.is_present("r")) {
+        args_parser.print_usage();
+        return -1;
+    }
+
+    if (args.is_present("s") && args.is_present("r")) {
+        args_parser.print_usage();
+        return -1;
+    }
+
+    auto op = OP_Set;
+    int size = 0;
+
+    if (args.is_present("s")) {
+        auto str = args.get("s");
+
+        switch (str[0]) {
+        case '+':
+            op = OP_Grow;
+            str = str.substring(1, str.length() - 1);
+            break;
+        case '-':
+            op = OP_Shrink;
+            str = str.substring(1, str.length() - 1);
+            break;
+        }
+
+        bool ok;
+        size = str.to_int(ok);
+        if (!ok) {
+            args_parser.print_usage();
+            return -1;
+        }
+    }
+
+    if (args.is_present("r")) {
+        struct stat st;
+        int rc = stat(args.get("r").characters(), &st);
+        if (rc < 0) {
+            perror("stat");
+            return -1;
+        }
+
+        op = OP_Set;
+        size = st.st_size;
+    }
+
+    auto name = args.get_single_values()[0];
+
+    int fd = open(name.characters(), O_CREAT, O_RDWR);
```

Comment:
Oops, didn't spot this before:
The 3rd argument to `open()` is the file mode (only used with `O_CREAT`)
So you probably want something more like:
```cpp
int fd = open(name.characters(), O_CREAT | O_RDWR, 0666);
```